src/Features/Core/Portable/CodeFixesAndRefactorings/CodeActionRequestPriorityProvider.cs

```diff
@@ -0,0 +1,146 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Microsoft.CodeAnalysis.CodeActions
+{
+    internal sealed class CodeActionRequestPriorityProvider
+    {
+        /// <summary>
+        /// Set of analyzers which have been de-prioritized to <see cref="CodeActionRequestPriority.Low"/> bucket.
+        /// </summary>
+        private readonly HashSet<DiagnosticAnalyzer> _lowPriorityAnalyzers;
+
+        /// <summary>
+        /// Default provider with <see cref="CodeActionRequestPriority.None"/> <see cref="Priority"/>.
+        /// </summary>
+        public static readonly CodeActionRequestPriorityProvider Default = new(CodeActionRequestPriority.None, new());
```

Comment:
it worries me that we have static-readonlies that point at objects with mutable state (HashSets are mutable).  If this was an immutablehashset i'd feel better.